uktrade/data-hub-components
# Add search result onClick event to entity search component

## Description of change
Adds the entity search result click event to the entity search component.

## Test instructions
1. Browse to `/?path=/story/entitysearch--data-hub-company-search`
2. Click `Search`
3. Click a search result. You should see an `alert`.

## Screenshots
There should be no visual change although in Storybook the click event will `alert` the entity as a string.
